pkg/crc/errors/multierror.go

```diff
@@ -22,12 +22,30 @@ func (m MultiError) ToError() error {
 	if len(m.Errors) == 0 {
 		return nil
 	}
+	if len(m.Errors) == 1 {
+		return m.Errors[0]
+	}
+
+	var aggregatedErrors []string
 
-	errStrings := []string{}
-	for _, err := range m.Errors {
-		errStrings = append(errStrings, err.Error())
+	count := 1
+	current := m.Errors[0].Error()
+	for i := 1; i < len(m.Errors); i++ {
+		if m.Errors[i].Error() == current {
+			count++
+			continue
+		}
+		aggregatedErrors = append(aggregatedErrors, errorWithCount(current, count))
+		count = 1
+		current = m.Errors[i].Error()
 	}
-	return fmt.Errorf(strings.Join(errStrings, "\n"))
+	aggregatedErrors = append(aggregatedErrors, errorWithCount(current, count))
```

Comment:
I think I'd use a `strings.Builder` rather than a `[]string` + `append`/`strings.Join()`

Reply:
in the background of `Join` implementation, it does uses the `Builder` itself.
```golang
// Join concatenates the elements of its first argument to create a single string. The separator
// string sep is placed between elements in the resulting string.
func Join(elems []string, sep string) string {
	switch len(elems) {
	case 0:
		return ""
	case 1:
		return elems[0]
	}
	n := len(sep) * (len(elems) - 1)
	for i := 0; i < len(elems); i++ {
		n += len(elems[i])
	}

	var b Builder
	b.Grow(n)
	b.WriteString(elems[0])
	for _, s := range elems[1:] {
		b.WriteString(sep)
		b.WriteString(s)
	}
	return b.String()
}
```